dbc_teacher.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'dbc_member'
+
+class DbcTeacher < DbcMember
+  attr_reader :salary, :target_raise
+  def initialize(options={})
+    super
+    @phase = 3
+  end
+
+  def set_phase(num)
+    @phase = num
+    "Cool, I've always wanted to teach phase #{num}!"
+  end
+
+  def salary=(new_salary)
+    puts "This better be good!"
+    @salary = new_salary
+  end
+
+  def receive_raise(raise)
+    @salary += raise
+  end
+
+  def set_performance_rating(rating)
+    response = ""
+    if rating > 90 && self.class == Teacher
+      receive_raise(@target_raise)
+      response = "Yay, I'm a great employee!"
+    elsif rating > 80 && self.class == ApprenticeTeacher
```

Comment:
Any time you find yourself checking if your class is a specific subclass, that's a big red flag that you're not doing polymorphism the ruby way.  Instead, define these values in the subclasses, either as instance variables or constants, and then you don't have to special case it here.